extra/modules/fiftyone-devicedetection/src/main/java/org/prebid/server/hooks/modules/fiftyone/devicedetection/config/FiftyOneDeviceDetectionModuleConfiguration.java

```diff
@@ -0,0 +1,49 @@
+package org.prebid.server.hooks.modules.fiftyone.devicedetection.config;
+
+import fiftyone.devicedetection.DeviceDetectionPipelineBuilder;
+import fiftyone.pipeline.core.flowelements.Pipeline;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.FiftyOneDeviceDetectionModule;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.core.DeviceEnricher;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.core.PipelineBuilder;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.hooks.FiftyOneDeviceDetectionEntrypointHook;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.v1.hooks.FiftyOneDeviceDetectionRawAuctionRequestHook;
+import org.prebid.server.hooks.modules.fiftyone.devicedetection.model.config.ModuleConfig;
+import org.prebid.server.hooks.v1.Hook;
+import org.prebid.server.hooks.v1.InvocationContext;
+import org.prebid.server.hooks.v1.Module;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import java.util.Set;
+
+@Configuration
+@ConditionalOnProperty(prefix = "hooks." + FiftyOneDeviceDetectionModule.CODE, name = "enabled", havingValue = "true")
```

Comment:
I know you have seen this approach inside the Confiant module, but we are going to remove it in the next major PBS release (PBS 4.0). The main idea is not to require a separate config file for the module. Please update the property handling approach to be similar to other modules (not Confiant), for example, richmedia-filter.

Reply:
Hi @SerhiiNahornyi 
Thanks for these comments, the previous ones were addressed, however this one we are a bit confused about. 

1. We see this same pattern in both ortb2blocking and richmedia-filter and 
* https://github.com/prebid/prebid-server-java/blob/master/extra/modules/ortb2-blocking/src/main/java/org/prebid/server/hooks/modules/ortb2/blocking/spring/config/Ortb2BlockingModuleConfiguration.java#L10
* https://github.com/prebid/prebid-server-java/blob/master/extra/modules/pb-richmedia-filter/src/main/java/org/prebid/server/hooks/modules/pb/richmedia/filter/config/PbRichmediaFilterModuleConfiguration.java#L17

2.  Also [Prebid Server Java Modularity Tech Spec](https://docs.google.com/document/d/1VP_pi7L5Iy3ikHMbtC2_rD5RZTVSc3OkTWKvtRS5x5Y/edit) says this: 

> If a module is disabled, its implementation must not be present in a Spring context. This could be achieved by annotating the module's Spring configuration with
```
@ConditionalOnProperty(prefix = "hooks.MODULECODE", name = "enabled", havingValue = "true").
```

3. > The main idea is not to require a separate config file for the module.
I don't think we require a separate config file - all settings live in the server config yaml.  [f.e.](https://github.com/51Degrees/prebid-server-java/blob/add-51degrees-device-detection-module/sample/prebid-config-with-51d-dd.yaml)

Could you please elaborate how this should be updated?   Thanks.